### PR TITLE
KTOR-1338 Add ktlint build configuration

### PR DESCRIPTION
**Pull request by cy6erGn0m**
Should be merged after https://github.com/ktorio/ktor/pull/2208  (merged)

This adds a new build configuration to run [ktlint](https://github.com/pinterest/ktlint) only without build and show reports

### CODE REVIEW

File: .teamcity/src/subprojects/build/core/CodeStyleVerifyBuild.kt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    vcs {
+        root(VCSCore)
+    }
+
+    steps {
+        gradle {
+            name = "Run ktlint"
+            tasks = "lintKotlin"
+            gradleParams = "-PenableCodeStyle=true"
+            jdkHome = "%env.${jdkVersions.last().env}%"
```
Comment (hhariri) [inline .teamcity/src/subprojects/build/core/CodeStyleVerifyBuild.kt:25]: A reason for being this and not explicitly a version?

File: .teamcity/src/subprojects/build/core/CodeStyleVerifyBuild.kt [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        }
+    }
+
+    triggers {
+        vcs {
+            // we only verify kts so no need to trigger it every time
+            triggerRules = listOf(
```
Comment (hhariri) [inline .teamcity/src/subprojects/build/core/CodeStyleVerifyBuild.kt:32]: String literals maybe?

File: .teamcity/src/subprojects/build/core/CodeStyleVerifyBuild.kt (hunk 1 above)
Comment (cy6erGn0m) [reply .teamcity/src/subprojects/build/core/CodeStyleVerifyBuild.kt:25]: Makes no big difference. Can apply `java11` to be sure.

File: .teamcity/src/subprojects/build/core/CodeStyleVerifyBuild.kt (hunk 2 above)
Comment (cy6erGn0m) [reply .teamcity/src/subprojects/build/core/CodeStyleVerifyBuild.kt:32]: multiline with trimIndent?

**Comment by cy6erGn0m** [comment]
Fixed and added more trigger rules to cover editorconfig changes and more gradle changes

**Review by hhariri — APPROVED** [review]

**Pull request merged**